components/devtools/lib.rs

```diff
@@ -479,31 +488,43 @@ impl DevtoolsInstance {
     fn handle_network_event(
         &mut self,
         connections: Vec<TcpStream>,
-        pipeline_id: PipelineId,
         request_id: String,
         network_event: NetworkEvent,
     ) {
-        let netevent_actor_name = self.find_network_event_actor(request_id);
-
-        let Some(id) = self.pipelines.get(&pipeline_id) else {
-            return;
+        let browsing_context_id = match &network_event {
+            NetworkEvent::HttpRequest(req) => Some(req.browsing_context_id),
+            NetworkEvent::HttpResponse(resp) => Some(resp.browsing_context_id),
         };
-        let Some(browsing_context_actor_name) = self.browsing_contexts.get(id) else {
+
+        let watcher_name = browsing_context_id
+            .and_then(|id| self.browsing_contexts.get(&id))
+            .map(|actor_name| {
+                self.actors
+                    .lock()
+                    .unwrap()
+                    .find::<BrowsingContextActor>(actor_name)
+                    .watcher
+                    .clone()
+            });
+            
+        if watcher_name.is_none() {
```

Comment:
Let's do:
```rust
let Some(browsing_context_actor_name) = self.browsing_contexts.get(&browsing_context_id) else { return };
let watcher_name = self.actors
    .lock()
    .unwrap()
    .find::<BrowsingContextActor>(browsing_context_actor_name)
    .watcher
   .clone();
```
Then there are no unwraps required and it's clear where any failure needs to be handled.